core/schemas/bifrost.go

```diff
@@ -142,24 +147,74 @@ type ToolChoice struct {
 // BifrostMessage represents a message in a chat conversation.
 type BifrostMessage struct {
 	Role    ModelChatMessageRole `json:"role"`
-	Content *string              `json:"content,omitempty"`
+	Content MessageContent       `json:"content"`
 
 	// Embedded pointer structs - when non-nil, their exported fields are flattened into the top-level JSON object
 	// IMPORTANT: Only one of the following can be non-nil at a time, otherwise the JSON marshalling will override the common fields
-	*UserMessage
 	*ToolMessage
 	*AssistantMessage
 }
 
-// UserMessage represents a message from a user
-type UserMessage struct {
-	ImageContent *ImageContent `json:"image_content,omitempty"`
+type MessageContent struct {
```

Comment:
@coderabbitai by any means, can we use generics here to avoid dancing around embedded objects?